addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js

```diff
@@ -530,11 +530,16 @@ function factory(dependencies) {
             // 1. move messages from inbox to history
             // AKU TODO: flag other caches to invalidate
             // task-2171873
-            inboxMailbox.messages.map(message => {
-                message.update({
-                    isNeedaction: false,
-                    isHistory: true,
-                });
+            message_ids.map(message_id => {
+                const message = inboxMailbox.messages.find(
+                    inboxMessage => inboxMessage.id === message_id
+                );
+                if (message) {
+                    message.update({
+                        isNeedaction: false,
+                        isHistory: true,
+                    });
+                }
```

Comment:
Double check how reliable `message_ids` is in contexts of 1) mark single inbox message as read 2) mark all inbox messages as read. I think at some point mark as read notification only considered the "mark all inbox messages as read" use-case, but now it looks like it could also be single inbox message mark as read.

So looks like previous code of handlers was wrong.

Reply:
`mark_all_as_read` -> `message_ids` is filled with all the messages that have been marked as read.
Note that this whole method can potentially be very bad for performance if there are a crazy amount of unread messages.

`set_message_done` -> `message_ids` is also filled with all the messages
In this case, all related `channel_ids` are also sent, this is why there is the flux specific thing about channels in the handler.
Note: `message_needaction_counter` is also on thread and not just on channels so this logic is flawed, but it's out scope. (`message_needaction_counter` should probably be (re)moved completely, as there was already a task merged in master to remove it from chatter threads but it was kept for channels)

-> `message_ids` appears to be reliable in all cases.